docs/docs/contribute/software/add-new-metric-provider.md

```diff
@@ -0,0 +1,140 @@
+   in the `metrics-operator/controllers/common/providers/provider.go` file.
+   add a case for the `KeptnPlaceholderProviderType`.
+   Instantiate the placeholder provider struct and return it.
+
+    ```go
+    // Inside the providers package
+
+    // NewProvider function
+    func NewProvider(providerType string, log logr.Logger, k8sClient client.Client) (KeptnSLIProvider, error) {
+        switch strings.ToLower(providerType) {
+        case KeptnPlaceholderProviderType:
+            return &placeholder.KeptnPlaceholderProvider{
+                Log:        log,
+                HttpClient: http.Client{},
+            }, nil
+        // Other cases...
+        }
+    }
+    ```
+
+5. **Update the validation webhook and crd config:** To update the validation webhook and crd config of the metrics operator.
+   Add the provider name next to last providers on this
+   [line](https://github.com/keptn/lifecycle-toolkit/blob/main/metrics-operator/api/v1beta1/keptnmetricsprovider_types.go#L29)
+   to look like this
+
+    `// +kubebuilder:validation:Pattern:=prometheus|dynatrace|datadog|dql|placeholder`.
+
+     In the metric-operator directory run `make manifests` to update the metrics-operator crd config
+     Then modify the helm chart and the helm chart crd validation to match the update in the metrics-operator crd config
+  
+6. **Add Test Cases:**
+     * Write a unit test to validate your implementation at the function level.
+       Unit tests ensure that individual
+       functions behave as expected and meet their functional requirements.
+
+     * Include a Chainsaw test to validate the behavior of Kubernetes resources managed by your code.
+       Chainsaw tests simulate real-world scenarios and interactions within a Kubernetes cluster, ensuring
+       the correctness of your Kubernetes configurations and deployments.
+
+         **NB:** For testing the metricsprovider a single KeptnMetric and KeptnMetricsProvider is needed
+
+        Below are the steps for adding an integration test.
+
+        * In the directory `test/chainsaw/testmetrics`, create a folder `keptn-metrics-validation` in our case.
+        * Within the `keptn-metrics-validation` folder, create three YAML files `00-install.yaml`, `goodmetrics.yaml`
+          and `badmetrics.yaml`.
+          * `00-install.yaml` contains a sample configuration that installs a valid `KeptnMetricsProvider`
+              in our case `placeholder`.
+          * `goodmetrics.yaml` define a sample `KeptnMetric` configuration representing a valid use case, while
+          * `badmetrics.yaml` define a sample `KeptnMetric` configuration containing errors or incorrect values.
+        * Create a file named `chainsaw-test.yaml` and define the steps for the integration test in chainsaw-test.yaml.
+
+        > For more information checkout [an already existing integration test](https://github.com/keptn/lifecycle-toolkit/tree/main/test/chainsaw/testmetrics/metrics)
```

Comment:
This is not the proper test that should be linked here. In the same folder we have a test called `metrics-provider` which is more suitable

Reply:
Alright I will update the doc